src/validation.cpp

```diff
@@ -2075,15 +2076,18 @@ bool CChainState::ConnectBlock(const CBlock& block, BlockValidationState& state,
 
     CBlockUndo blockundo;
 
+    // Precomputed transaction data pointers must not be invalidated
+    // until after control has run the script checks (potentially
+    // in multiple threads). Keep txsdata in scope for as long as
+    // control.
     CCheckQueueControl<CScriptCheck> control(fScriptChecks && g_parallel_script_checks ? &scriptcheckqueue : nullptr);
+    std::vector<PrecomputedTransactionData> txsdata(block.vtx.size());
```

Comment:
`std::array<PrecomputedTransactionData, block.vtx.size()> txsdata;` make invalidation via resizing impossible now that it's not being appended to(I think?)